pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -204,15 +205,21 @@ func (c *Controller) removeStaleGatewayRoutes() error {
 	}
 
 	// TODO: This is not the best place to keep the ClusterIP Service routes.
-	desiredClusterIPSvcIPs := map[string]bool{}
+	desiredClusterIPSvcIPs := sets.NewString()
 	if c.proxyAll && runtime.IsWindowsPlatform() {
 		// The route for virtual IP -> antrea-gw0 should be always kept.
-		desiredClusterIPSvcIPs[config.VirtualServiceIPv4.String()] = true
+		vIPNet := util.NewIPNet(config.VirtualServiceIPv4).String()
+		desiredClusterIPSvcIPs.Insert(vIPNet)
 
 		svcs, err := c.svcLister.List(labels.Everything())
 		for _, svc := range svcs {
-			for _, ip := range svc.Spec.ClusterIPs {
-				desiredClusterIPSvcIPs[ip] = true
+			for _, clusterIP := range svc.Spec.ClusterIPs {
+				clusterIPNet := util.NewIPNet(net.ParseIP(clusterIP)).String()
+				desiredClusterIPSvcIPs.Insert(clusterIPNet)
+			}
+			for _, ingressIP := range svc.Status.LoadBalancer.Ingress {
+				ingressIPNet := util.NewIPNet(net.ParseIP(ingressIP.IP)).String()
+				desiredClusterIPSvcIPs.Insert(ingressIPNet)
```

Comment:
It would cause potential issue by appending the ingress IPs to a set for cluster IPs. For example, it could cause the calculated ClusterIP CIDR much bigger than actual value.
If the needed set is supposed to contain both IPs, please rename the variable.

Reply:
This is only for Windows, Linux ClusterIP CIDR is not calculated from this.

Reply:
By potential issue, I meant it could cause a bug in future if others think the set contains pure cluster IPs, which is mislead by the name.